File: app/src/main/java/com/sourcepoint/example_app/MainActivity.java
package com.sourcepoint.example_app;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.ViewGroup;

import androidx.appcompat.app.AppCompatActivity;
import com.sourcepoint.gdpr_cmplibrary.GDPRConsentLib;
import com.sourcepoint.gdpr_cmplibrary.NativeMessage;
import com.sourcepoint.gdpr_cmplibrary.NativeMessageAttrs;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;
import java.util.UUID;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "**MainActivity";

    private ViewGroup mainViewGroup;

    private PropertyConfig config;

    private void showView(View view) {
        if(view.getParent() == null){
            view.setLayoutParams(new ViewGroup.LayoutParams(0, 0));
            view.getLayoutParams().height = ViewGroup.LayoutParams.MATCH_PARENT;
            view.getLayoutParams().width = ViewGroup.LayoutParams.MATCH_PARENT;
            view.bringToFront();
            view.requestLayout();
            mainViewGroup.addView(view);
        }
    }
    private void removeView(View view) {
        if(view.getParent() != null)
            mainViewGroup.removeView(view);
    }

    private PropertyConfig getConfig(int configResource){
        PropertyConfig config = null;
        try {
            config = new PropertyConfig(new JSONObject(new Scanner(getResources().openRawResource(configResource)).useDelimiter("\\A").next()));
        } catch (JSONException e) {
            Log.e(TAG, "Unable to parse config file.", e);
        }
        return config;
    }

    private GDPRConsentLib buildGDPRConsentLib() {

        SharedPreferences sharedPref = this.getSharedPreferences("myshared", Context.MODE_PRIVATE);
        if(!sharedPref.contains("MyAppsAuthId")){
            String uniqueID = UUID.randomUUID().toString();
            System.out.println("cookie ["+ uniqueID +"]");
            sharedPref.edit().putString("MyAppsAuthId", uniqueID).apply();
        }

        return GDPRConsentLib.newBuilder(config.accountId, config.propertyName, config.propertyId, config.pmId,this)
                .setOnConsentUIReady(view -> {
                    showView(view);
                    Log.i(TAG, "onConsentUIReady");
                })
                .setOnConsentUIFinished(view -> {
                    removeView(view);
                    Log.i(TAG, "onConsentUIFinished");
                })
                .setOnConsentReady(consent -> {
                    Log.i(TAG, "onConsentReady");
                    Log.i(TAG, "uuid: " + consent.uuid );
                    Log.i(TAG, "consentString: " + consent.consentString);
                    Log.i(TAG, "TCData: " + consent.TCData);
                    Log.i(TAG, "vendorGrants: " + consent.vendorGrants);
                    for (String vendorId : consent.acceptedVendors) {
                        Log.i(TAG, "The vendor " + vendorId + " was accepted.");
                    }
                    for (String purposeId : consent.acceptedCategories) {
                        Log.i(TAG, "The category " + purposeId + " was accepted.");
                    }
                    for (String purposeId : consent.legIntCategories) {
                        Log.i(TAG, "The legIntCategory " + purposeId + " was accepted.");
                    }
                    for (String specialFeatureId : consent.specialFeatures) {
                        Log.i(TAG, "The specialFeature " + specialFeatureId + " was accepted.");
                    }
                })
                .setOnError(error -> {
                    Log.e(TAG, "Something went wrong: ", error);
                    Log.e(TAG, "ConsentLibErrorMessage: " + error.consentLibErrorMessage);
                })
                .setOnPMReady(()-> Log.i(TAG, "PM Ready"))
                .setOnMessageReady(()-> Log.i(TAG, "Message Ready"))
                .setOnPMFinished(()-> Log.i(TAG, "PM Finished"))
                .setOnMessageFinished(()-> Log.i(TAG, "Message Finished"))
                .setOnAction(actionType  -> Log.i(TAG , "ActionType : "+actionType.toString()))
                .setAuthId(sharedPref.getString("MyAppsAuthId",""))
                .build();
    }

    private NativeMessage buildNativeMessage(){
        return new NativeMessage(this){
            @Override
            public void init(){
                super.init();
                // When using a customized layout one can completely override the init method
                // not calling super.init() and inflating the native view with the chosen layout instead.
                // In this case its important to set all the default child views using the setter methods
                // like its done in the super.init()
            }
            @Override
            public void setAttributes(NativeMessageAttrs attrs){
                super.setAttributes(attrs);
                //Here one can extend this method in order to set customized attributes other then the ones
                //already set in the super.setAttributes. No need to completely override this method.
            }
        };
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.i(TAG, "init");
        buildGDPRConsentLib().run();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mainViewGroup = findViewById(android.R.id.content);
        config = getConfig(R.raw.mobile_demo_web);
        findViewById(R.id.review_consents).setOnClickListener(_v -> buildGDPRConsentLib().showPm());
        findViewById(R.id.open_activity).setOnClickListener(_v -> startActivity(new Intent(this, MainActivityAuthId.class)));
    }
}

File: samples/nativemessage/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.os.Bundle
import android.util.Log
import com.google.android.material.floatingactionbutton.FloatingActionButton
import com.google.android.material.snackbar.Snackbar
import androidx.appcompat.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import com.sourcepoint.gdpr_cmplibrary.GDPRConsentLib
import com.sourcepoint.gdpr_cmplibrary.NativeMessage
import com.sourcepoint.gdpr_cmplibrary.NativeMessageAttrs
import kotlinx.android.synthetic.main.content_main.*

class MainActivity : AppCompatActivity() {

    private val mainViewGroup by lazy<ViewGroup> { findViewById(android.R.id.content) }

    private val TAG = "**MainActivity"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(findViewById(R.id.toolbar))

        findViewById<FloatingActionButton>(R.id.fab).setOnClickListener { view ->
            Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                .setAction("Action", null).show()
        }

        consent.setOnClickListener {
            buildGDPRConsentLib().showPm()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        return when (item.itemId) {
            R.id.action_settings -> true
            else -> super.onOptionsItemSelected(item)
        }
    }

    override fun onResume() {
        super.onResume()
        Log.i(TAG, "init");
        buildGDPRConsentLib().run(buildNativeMessage());
    }

    fun buildGDPRConsentLib(): GDPRConsentLib {
        return GDPRConsentLib
            .newBuilder(
                Accounts.nativeAccount.accountId,
                Accounts.nativeAccount.propertyName,
                Accounts.nativeAccount.propertyId,
                Accounts.nativeAccount.pmId,
                applicationContext
            )
            .setOnConsentUIReady { view ->
                showView(view);
                Log.i(TAG, "onConsentUIReady");
            }
            .setOnConsentUIFinished { view ->
                removeView(view);
                Log.i(TAG, "onConsentUIFinished");
            }
            .setOnConsentReady { consent ->
                Log.i(TAG, "onConsentReady")
                Log.i(TAG, "consentString: " + consent.consentString)
                Log.i(TAG, consent.TCData.toString())
                for (vendorId in consent.acceptedVendors) {
                    Log.i(TAG, "The vendor $vendorId was accepted.")
                }
                for (purposeId in consent.acceptedCategories) {
                    Log.i(TAG, "The category $purposeId was accepted.")
                }
                for (purposeId in consent.legIntCategories) {
                    Log.i(TAG, "The legIntCategory $purposeId was accepted.")
                }
                for (specialFeatureId in consent.specialFeatures) {
                    Log.i(
                        TAG,
                        "The specialFeature $specialFeatureId was accepted."
                    )
                }
            }
            .setOnError { error ->
                Log.e(TAG, "Something went wrong: ", error)
                Log.i(TAG, "ConsentLibErrorMessage: " + error.consentLibErrorMessage)
            }
            .build()
    }

    private fun removeView(view: View) {
        if (view.parent != null) mainViewGroup.removeView(view)
    }

    private fun showView(view: View) {
        if (view.parent == null) {
            view.layoutParams = ViewGroup.LayoutParams(0, 0)
            view.layoutParams.height = ViewGroup.LayoutParams.MATCH_PARENT
            view.layoutParams.width = ViewGroup.LayoutParams.MATCH_PARENT
            view.bringToFront()
            view.requestLayout()
            mainViewGroup.addView(view)
        }
    }



    private fun buildNativeMessage(): NativeMessage? {
        return layoutInflater.inflate(R.layout.custom_layout, null) as? NativeMessage
    }
}